packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -384,15 +391,24 @@ const QueryEditorHeaderContent = observer(
             <PanelListItem>
               <div className="input--with-validation">
                 <input
-                  title="Query title rename"
                   ref={renameRef}
                   className={clsx('input input--dark', {
                     'input--caution': isExistingQueryName,
                   })}
                   onChange={changeQueryName}
-                  onKeyDown={onEnter}
+                  onKeyDown={(event) => {
+                    if (event.code === 'Enter') {
+                      event.stopPropagation();
+                      setShowQueryNameEditInput(false);
+                      onEnter(event);
+                    } else if (event.code === 'Escape') {
+                      setShowQueryNameEditInput(false);
+                      event.stopPropagation();
+                      renameState.renameQueryState.reset();
+                      renameState.editorStore.setRenameState(undefined);
+                    }
+                  }}
                   value={renameState.queryName}
-                  placeholder="Search"
                 />
                 {isExistingQueryName && (
```

Comment:
Did we leverage the API you added in https://github.com/finos/legend-engine/pull/1669 ?